login-module/src/main/java/com/wipro/ats/bdre/security/SessionCheckPerRequestFilter.java

```diff
@@ -98,7 +100,7 @@ public boolean isTrusted(X509Certificate[] certificate, String authType) {
             AuthResult authResult = null;
             try {
                 authResult = restTemplate.getForObject(completeServiceURL + token, AuthResult.class);
-            } catch (Exception e) {
+            } catch (BDREException e) {
                 LOGGER.error(e);
                 throw new RuntimeException("Error Occurred. Service URL:" + completeServiceURL + token + "; servletRequest.isSecure()=" + servletRequest.isSecure(), e);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS00112)